crates/ra_syntax/src/parsing/text_token_source.rs

```diff
@@ -48,7 +48,7 @@ impl<'t> TokenSource for TextTokenSource<'t> {
 
     fn is_keyword(&self, kw: &str) -> bool {
         let pos = self.curr.1;
-        if !(pos < self.tokens.len()) {
```

Comment:
This is super counter-intuitive, but I actually find the previous variant easier to understand. I never use `>` or `>=`, because `< `and `<=` have nice geometric intuition with a number line. For checking indexes specifically, `idx < len` is an invariant, and `!(idx < len)` reads more like "invariant does not not hold" than `idx >= len`, where you need to mentaly flip both the direction and strictness